src/main/java/it/gov/pagopa/paymentupdater/service/PaymentServiceImpl.java

```diff
@@ -132,4 +127,15 @@ public Optional<Payment> findById(String messageId) {
 		return paymentRepository.findById(messageId);
 	}
 
+	@Override
+	public List<Payment> getPaymentsByRptid(String rptid) {
+		List<Payment> payments = paymentRepository.getPaymentByRptId(rptid);
+		return payments == null ? new ArrayList<>() : payments;
+	}
+
+	@Override
+	public int countFindById(String id) {
```

Comment:
ditto

Reply:
As mentioned, if a duplicate message arrives, we check the existence of the payment. Could this check affect performance?

Reply:
fixed